engine/src/main/java/org/pentaho/di/trans/streaming/common/BaseStreamStepMeta.java

```diff
@@ -75,20 +80,44 @@
     BeanInjectionInfo info = new BeanInjectionInfo( this.getClass() );
     BeanInjector injector = new BeanInjector( info );
     Map<String, BeanInjectionInfo.Property> properties = info.getProperties();
+
+    StringBuilder buider = new StringBuilder();
+
+    info.getGroups()
+      .forEach( group -> {
+        boolean groupHasName = !isNullOrEmpty( group.getName() );
```

Comment:
Can we move this lambda to a method and use a method reference?  It's pretty big for inline lambda.  Giving it a name will help understand what it's doing too.

Reply:
Good idea.